core/src/muxing.rs

```diff
@@ -91,14 +91,19 @@ pub trait StreamMuxer {
     /// is ready to be polled, similar to the API of `Stream::poll()`.
     /// Only the latest task that was used to call this method may be notified.
     ///
+    /// Implementors may rely on this method being called soon after the registered
+    /// task has been notified. It is thus possible to use this method to perform
+    /// background work, such as processing incoming packets and polling timers.
```

Comment:
Why do we need to commit to this? The PR title is "Improve muxer docs", but this bit promises specific API guarantees that have not been discussed anywhere.

Reply:
@tomaka has told me this privately, and his work on QUIC support requires it.  More generally, any muxer that needs to have background work done needs this.

Reply:
My own implementation of QUIC support did not need it, but had to spawn a background task for each muxer, which is unfortunate.